graphbolt/src/csc_sampling_graph.cc

```diff
@@ -257,17 +259,23 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighborsImpl(
                   continue;
                 }
 
+                // Pre-allocate tensors for each node. Because the pick
+                // functions are modified, this part of code needed refactoring
+                // to adapt to the change of APIs. It's temporary since the
+                // whole process will be rewritten soon.
```

Comment:
When you say whole process, better to clarify what process, it is a little bit confusing when I read this.
Just to confirm, the process you mean is preallocate space for picked_neighbors for all threads?

Reply:
Yes, the function `SampleNeighborsImpl` which contains the main logic of the sampling process will be rewritten in the next pr. It will pre-allocate the final tensor instead of allocating separately and thus eliminate `torch::cat` operation.